### PR TITLE
feat: reusable smart contracts package

### PR DESCRIPTION
**Pull request by r0x4xn**
## Description
Includes a new package `prepo-smart-contracts` that can be used across the `smart-contracts` projects to import reusable smart contracts

### CODE REVIEW

**Comment by Nilay27** [comment]
Files for SafeOwnable  are also added (copied from[ prepo-nft-game-private](https://github.com/prepo-io/prepo-nft-game-private/blob/master/acquisition-royale/contracts/SafeOwnable.sol))

These are to be used for all the ownable contracts in core as well as token projects to be switched to a safer version of nominate + accept method.

**Note**: As the files were already completed tested and merged hence submitted all of them at once.

File: packages/prepo-smart-contracts/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "prepo-smart-contracts",
+  "version": "0.0.0",
+  "description": "prePO Reusable smart contracts",
```
Comment (XavierEkkel) [inline packages/prepo-smart-contracts/package.json:4]:
```suggestion
  "description": "prePO Shared Smart Contracts",
```

File: packages/prepo-smart-contracts/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "prepo-smart-contracts",
```
Comment (XavierEkkel) [inline packages/prepo-smart-contracts/package.json:2]:
```suggestion
  "name": "prepo-shared-contracts",
```

File: packages/prepo-smart-contracts/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# prePO reusable smart-contracts
```
Comment (XavierEkkel) [inline packages/prepo-smart-contracts/README.md:1]:
```suggestion
# prePO Shared Smart Contracts
```

File: apps/smart-contracts/token/contracts/TokenShop.sol [hunk 4]
```diff
@@ -9,6 +9,7 @@ import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
 import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
 import "./interfaces/ITokenShop.sol";
 import "./interfaces/IPurchaseHook.sol";
+// import "prepo-smart-contracts/contracts/SafeOwnable.sol"; // Example on how to use the prepo-smart-contracts package
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/TokenShop.sol:12]:
```suggestion
// import "prepo-shared-contracts/contracts/SafeOwnable.sol"; // Example on how to use the prepo-shared-contracts package
```

**Review by XavierEkkel — APPROVED** [review]
LGTM after renaming the directory to `prepo-shared-contracts` (more concise that `prepo-shared-smart-contracts` and more accurate than `prepo-smart-contracts`) and adding `.solhint.json`, `.solhintignore`, `.solcover.js` to the `packages/prepo-shared-contracts` directory

**Pull request merged**